go/upgrade/api/api.go

```diff
@@ -0,0 +1,114 @@
+// Package api defines the interface exporting the upgrade infrastructure's functionality.
+package api
+
+import (
+	"context"
+
+	"github.com/oasislabs/oasis-core/go/common/errors"
+	epochtime "github.com/oasislabs/oasis-core/go/epochtime/api"
+)
+
+const (
+	// ModuleName is the upgrade module name.
+	ModuleName = "upgrade"
+
+	// UpgradeMethInternal is the internal upgrade method,
+	// where the node binary itself has the migration code.
+	UpgradeMethInternal = "internal"
+)
+
+// UpgradeStage is used in the upgrade descriptor to store completed stages.
+type UpgradeStage uint64
+
+const (
+	// UpgradeStageStartup is the startup upgrade stage, executed at the beginning of node startup.
+	UpgradeStageStartup UpgradeStage = 0
```

Comment:
Let's make zero an invalid upgrade stage as it is the default value. The first valid stage should start with one.